File: tests/pages/DashboardPage.js
export class DashboardPage {
  constructor(page) {
    this.page = page
    this.roleLabel = page.locator('strong')
    this.clientList = page.locator('ul')
    this.logoutButton = page.locator('button:has-text("Logout")')
  }

  async logout() {
    await this.logoutButton.click()
  }
}

File: tests_py/pages/dashboard_page.py
from playwright.sync_api import Page

class DashboardPage:
    """Authenticated home view — client list, role badge, logout."""

    def __init__(self, page: Page):
        self.page = page
        self.role_label = page.get_by_test_id("role-label")
        self.client_list = page.get_by_test_id("client-list")
        self.logout_button = page.get_by_role("button", name="Logout")

    def logout(self):
        self.logout_button.click()